evalml/tests/model_understanding_tests/test_graphs.py

```diff
@@ -60,46 +60,45 @@ def __init__(self, parameters):
     return TestPipeline(parameters={"Logistic Regression Classifier": {"n_jobs": 1}})
 
 
-@pytest.mark.parametrize("data_type", ['np', 'pd'])
-def test_confusion_matrix(data_type):
-    y_true = [2, 0, 2, 2, 0, 1]
-    y_predicted = [0, 0, 2, 2, 0, 2]
-    if data_type == 'pd':
-        y_true = pd.Series(y_true)
-        y_predicted = pd.Series(y_predicted)
+@pytest.mark.parametrize("data_type", ['np', 'pd', 'ww'])
+def test_confusion_matrix(data_type, make_data_type):
+    y_true = np.array([2, 0, 2, 2, 0, 1, 1, 0, 2])
+    y_predicted = np.array([0, 0, 2, 2, 0, 2, 1, 1, 1])
+    y_true = make_data_type(data_type, y_true)
+    y_predicted = make_data_type(data_type, y_predicted)
 
     conf_mat = confusion_matrix(y_true, y_predicted, normalize_method=None)
-    conf_mat_expected = np.array([[2, 0, 0], [0, 0, 1], [1, 0, 2]])
-    assert np.array_equal(conf_mat_expected, conf_mat)
+    conf_mat_expected = np.array([[2, 1, 0], [0, 1, 1], [1, 1, 2]])
+    assert np.array_equal(conf_mat_expected, conf_mat.to_numpy())
     assert isinstance(conf_mat, pd.DataFrame)
 
-    conf_mat = confusion_matrix(y_true, y_predicted, normalize_method='true')
-    conf_mat_expected = np.array([[1, 0, 0], [0, 0, 1], [1 / 3.0, 0, 2 / 3.0]])
-    assert np.array_equal(conf_mat_expected, conf_mat)
+    conf_mat = confusion_matrix(y_true, y_predicted, normalize_method='all')
+    conf_mat_expected = conf_mat_expected / 9.0
+    assert np.array_equal(conf_mat_expected, conf_mat.to_numpy())
     assert isinstance(conf_mat, pd.DataFrame)
 
-    conf_mat = confusion_matrix(y_true, y_predicted, normalize_method='pred')
-    conf_mat_expected = np.array([[2 / 3.0, np.nan, 0], [0, np.nan, 1 / 3.0], [1 / 3.0, np.nan, 2 / 3.0]])
```

Comment:
Not sure why this wasn't caught / if this was expected behavior? But based on our code for `confusion_matrix`, if any of the axis calculations while normalizing have a sum of zero, then we should raise an error (`test_normalize_confusion_matrix_error`); Updated values of matrix here so no error is thrown.

Reply:
There's a bug in the implementation/test 😬  The warning is only raised if the input to `normalize_confusion_matrix` is a numpy array but in `confusion_matrix` we always convert to a dataframe. This means there is no waning to catch so we don't raise the ValueError.

![image](https://user-images.githubusercontent.com/41651716/101946732-57a1a080-3bbd-11eb-90c0-b543b6ef205c.png)

Reply:
Ah you're absolutely right, it's only numpy that only displayed this warning 😢